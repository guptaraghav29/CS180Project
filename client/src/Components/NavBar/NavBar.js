import React from "react";

function changeStyleAnalytics() {
  //document.getElementById("analytics").style.cssText = 'background: #3B82F6; color: white;';
  document.getElementById("analytics").style.cssText =
    "border-bottom-color: #3B82F6;";
  //document.getElementById("bar").style.cssText = 'background-image: -webkit-linear-gradient(bottom, #3B82F6 -3px, transparent -3px, transparent 3px)';
  document.getElementById("search").style.cssText =
    "background: white; color: #3B82F6";
  document.getElementById("adv").style.cssText =
    "background: white; color: #3B82F6";
}
function changeStyleSearch() {
  //document.getElementById("search").style.cssText = 'background: #3B82F6; color: white';
  document.getElementById("search").style.cssText =
    "border-bottom-color: #3B82F6;";
  //document.getElementById("bar").style.cssText = 'background-image: -webkit-linear-gradient(bottom, #3B82F6 3px, transparent 3px, transparent 3px)';
  document.getElementById("analytics").style.cssText =
    "background: white; color: #3B82F6";
  document.getElementById("adv").style.cssText =
    "background: white; color: #3B82F6";
}
function changeStyleAdvSearch() {
  document.getElementById("search").style.cssText =
    "background: white; color: #3B82F6";
  document.getElementById("analytics").style.cssText =
    "background: white; color: #3B82F6";
  document.getElementById("adv").style.cssText =
    "border-bottom-color: #3B82F6;";
  //document.getElementById("bar").style.cssText = 'border-bottom-color: ';
  //document.getElementById("bar").style.cssText = 'background-image: -webkit-linear-gradient(bottom, #3B82F6 3px, transparent 3px, transparent 3px)';
  //document.getElementById("adv").style.cssText = 'background: #3B82F6; color: white';
  //linear-gradient(left, #3B82F6, #3B82F6; 70%, transparent 70%, transparent 100%); background-image: -webkit-linear-gradient(left, #3B82F6, #3B82F6 70%, transparent 70%, transparent 100%)
}

const NavBar = () => {
  return (
    <ul className="flex border-color-blue-500" id="bar">
      <li className="-mb-px mr-1">
        <a
          className="border-b-4 border-sky-blue-500 inline-block rounded-t py-2 px-4 text-blue-500 font-semibold hover:text-blue-800 font-semibold"
          id="search"
          href=""
          onClick={() => {
            changeStyleSearch();
          }}
        >
          Search
        </a>
      </li>
      <li className="mr-1">
        {/* <a
          className="border-b-4 border-sky-blue-500 inline-block rounded-t py-2 px-4 text-blue-500 font-semibold hover:text-blue-800 font-semibold"
          id="analytics"
          href="#analysis"
          onClick={() => {
            changeStyleAnalytics();
          }}
        >
          Analytics
        </a> */}
      </li>
      <li className="mr-1">
        {/* <a
          className="border-b-4 border-sky-blue-500 inline-block rounded-t py-2 px-4 text-blue-500 font-semibold hover:text-blue-800 font-semibold"
          id="adv"
          href="#adv"
          onClick={() => {
            changeStyleAdvSearch();
          }}
        >
          Advanced Search
        </a> */}
      </li>
    </ul>
  );
};

export default NavBar;
