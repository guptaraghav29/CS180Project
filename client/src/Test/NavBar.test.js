import * as React from "react";
import * as ReactDOM from "react-dom";
import NavBar from "../Components/NavBar/NavBar";
import renderer from "react-test-renderer";

import {
    fireEvent,
    getQueriesForElement,
    render,
} from "@testing-library/react";

it("testing the snapshot", () => {
	const tree = renderer.create(< NavBar />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("renders the search tab without any crashes", () => {
	const root = document.createElement("root");
	ReactDOM.render(< NavBar />, root);
	const { getByText } = getQueriesForElement(root);

	expect(getByText("Search")).not.toBeNull();
});

it("renders the probability tab without any crashes", () => {
	const root = document.createElement("root");
	ReactDOM.render(< NavBar />, root);
	const { getByText } = getQueriesForElement(root);

	expect(getByText("Probability")).not.toBeNull();
});


it("renders the advanced search tab without any crashes", () => {
	const root = document.createElement("root");
	ReactDOM.render(< NavBar />, root);
	const { getByText } = getQueriesForElement(root);

	expect(getByText("Advanced Search")).not.toBeNull();
});