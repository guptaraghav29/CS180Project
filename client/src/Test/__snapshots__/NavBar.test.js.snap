// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the snapshot 1`] = `
<ul
  className="flex border-color-blue-500"
  id="bar"
>
  <li
    className="-mb-px mr-1"
  >
    <a
      className="border-b-4 border-sky-blue-500 inline-block rounded-t py-2 px-4 text-blue-500 font-semibold hover:text-blue-800 font-semibold"
      href="#search"
      id="search"
      onClick={[Function]}
    >
      Search
    </a>
  </li>
  <li
    className="mr-1"
  >
    <a
      className="border-b-4 border-sky-blue-500 inline-block rounded-t py-2 px-4 text-blue-500 font-semibold hover:text-blue-800 font-semibold"
      href="#analysis"
      id="analytics"
      onClick={[Function]}
    >
      Analytics
    </a>
  </li>
  <li
    className="mr-1"
  >
    <a
      className="border-b-4 border-sky-blue-500 inline-block rounded-t py-2 px-4 text-blue-500 font-semibold hover:text-blue-800 font-semibold"
      href="#adv"
      id="adv"
      onClick={[Function]}
    >
      Advanced Search
    </a>
  </li>
</ul>
`;
